package/MDAnalysis/core/Selection.py

```diff
@@ -543,8 +518,7 @@ class ProteinSelection(Selection):
 
     .. SeeAlso:: :func:`MDAnalysis.lib.util.convert_aa_code`
     """
-    #: Dictionary of recognized residue names (3- or 4-letter).
-    prot_res = {x: None for x in [
+    prot_res = [
```

Comment:
On 7 Jan, 2016, at 12:54, Richard Gowers wrote:

> In package/MDAnalysis/core/Selection.py:
> 
> > @@ -543,8 +518,7 @@ class ProteinSelection(Selection):
> >  
> >      .. SeeAlso:: :func:`MDAnalysis.lib.util.convert_aa_code`
> >      """
> > -    #: Dictionary of recognized residue names (3- or 4-letter).
> > -    prot_res = {x: None for x in [
> > +    prot_res = [
> 
> I think the problem is .names or .resnames are going to be np arrays, so we're then using np.in1d which is numpy's version of doing this. And for that numpy wants another array.

So, in the future you do the selection by `np.in1d(ag.resnames, prot_res)`? That *might* be faster than `[resname in prot_res for resname in ag.resnames]` (although I'd be curious to see a benchmark). 

If you keep prot_res a list instead of a hash table then I would actually make prot_res a np.array right away to avoid the implicit asarray() inside np.in1p.